org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/json/JSONPreferenceInitializer.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.wildwebdeveloper.json;
+
+import org.eclipse.core.runtime.preferences.AbstractPreferenceInitializer;
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.wildwebdeveloper.Activator;
+
+import com.google.gson.JsonObject;
+
+public class JSONPreferenceInitializer extends AbstractPreferenceInitializer {
+	private static final IPreferenceStore STORE = Activator.getDefault().getPreferenceStore();
+	public static final String JSON_SCHEMA_PREFERENCE = "wildwebdeveloper.json.schema";
+
+	@Override
+	public void initializeDefaultPreferences() {
+		STORE.setDefault(JSON_SCHEMA_PREFERENCE, getDefaultJSONSchema());
+	}
+
+	private static String getDefaultJSONSchema() {
+		JsonObject schemaJson = new JsonObject();
+		schemaJson.addProperty("/.bower.json", "http://json.schemastore.org/bower");
```

Comment:
Shouldn't we declare those file names and patterns as content types (children of json contenttype) to the whole IDE, and then have the preference associate a content-type with a schema URL ?
I believe reusing the content type mechanisms allows more powerful features usually.